bin/spectacle/init.js

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const prompts = require('prompts');
+const parse = require('./args');
+
+const mdRegex = /(\.mdx$|\.md$)/gm;
+const jsRegex = /(\.json$|\.js$)/gm;
+const jsxRegex = /(\.jsx$|\.js$)/gm;
+
+const questions = [
+  {
+    type: 'select',
+    name: 'action',
+    message: 'Create a live server or build directory?',
+    choices: [
+      { title: 'Server', value: 'server' },
+      { title: 'Build', value: 'build' }
+    ]
+  },
+  {
+    type: prev => (prev === 'server' ? 'number' : null),
+    name: 'port',
+    message: 'What port for the live server?',
+    validate: val => {
+      const numLength = val.toString().length;
+      // eslint-disable-next-line no-magic-numbers
+      if (numLength === 4) {
+        return val;
+      }
+      return 'The port should be 4 numbers';
+    }
+  },
+  {
+    type: 'text',
+    name: 'src',
+    message:
+      'What is the filename of the markdown source? Include the file extension',
+    validate: name =>
+      name.match(mdRegex) ? true : 'The file must be a mdx or md file'
+  },
+  {
+    type: 'confirm',
+    name: 'custom_theme',
+    message: 'Do you have a custom theme file?',
+    initial: false
+  },
+  {
+    // eslint-disable-next-line no-constant-condition
+    type: prev => (prev ? 'text' : null),
+    name: 'theme',
+    message:
+      'What is the name of the custom theme file? Include the file extension',
+    validate: name =>
+      name.match(jsRegex) ? true : 'The file must be a js or json file'
+  },
+  {
+    type: 'confirm',
+    name: 'custom_template',
+    message: 'Do you have a custom template file?',
+    initial: false
+  },
+  {
+    // eslint-disable-next-line no-constant-condition
+    type: prev => (prev ? 'text' : null),
+    name: 'template',
+    message:
+      'What is the name of the custom template file? Include the file extension.',
+    validate: name =>
+      name.match(jsxRegex) ? true : 'The file must be a jsx or js file'
+  }
+];
+
+const promptArgs = (async() => {
```

Comment:
We need to not invoke this unconditionally, because we don't want to use the prompt unless we're in the "init mode".  Let's maybe just talk in `#spectacle` code channel about how to approach this from a CLI args perspective.

Reply:
Oh, just saw it in the ticket https://github.com/FormidableLabs/spectacle-cli/issues/51

If the user invokes with:

```sh
$ spectacle-cli init
```

We will use this functionality. If no `init` token is present, then we need to do the original "flags only" code path.

So what I think that means is:

1. Take this IIFE and just export a non-invoked `promptArgs` function.
2. When consuming in `bin/spectacle/args.js` figure out if `init` is in the CLI arguments. This will require likely a new `yargs` chain option.
3. If init, _then_ invoke `await promptArgs()`. If not, do old code flow.